<?xml version="1.0" ?>

<robot name="human_model" xmlns:xacro="http://www.ros.org/wiki/xacro">
  
  <link name="world"/>

  <!-- leg_ws to world -->
  <joint name="world_joint" type="fixed">
    <origin xyz="0 0 0" rpy="0 0 0.22"/>
    <parent link="world"/>
    <child link="leg_ws"/>
  </joint>

  <!-- leg_ws link -->
  <link name="leg_ws">
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://human_model/meshes/stl/leg_ws.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>

    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://human_model/meshes/stl/leg_ws.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"/>
    </visual>

    <inertial>
      <mass value="1"/>
      <inertia ixx="100"  ixy="0"  ixz="0" iyy="100" iyz="0" izz="100" />
    </inertial>
  </link>

  <!-- model to leg_ws -->
  <joint name="leg_joint" type="fixed">
    <origin xyz="-0.035 0 1.1" rpy="0 0 -0.22"/>
    <parent link="leg_ws"/>
    <child link="body"/>
  </joint>

  <!-- model link -->
  <link name="body">
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://human_model/meshes/stl/body.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>

    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://human_model/meshes/stl/body.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"/>
    </visual>

    <inertial>
      <mass value="1e-5" />
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6" />
    </inertial>
  </link>

  <!-- head to body -->
  <joint name="neck" type="fixed">
    <origin xyz="0 0 0.52" rpy="0 0 0"/>
    <parent link="body"/>
    <child link="head"/>
  </joint>

  <!-- head link -->
  <link name="head">
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://human_model/meshes/stl/head.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>

    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://human_model/meshes/stl/head.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"/>
    </visual>

    <inertial>
      <mass value="1e-5" />
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6" />
    </inertial>
  </link>

  <!-- =============================LEFT ARM========================================== -->
  <!-- upper to body -->
  <joint name="left_shoulder" type="fixed">
    <origin xyz="0.015 0.225 0.46" rpy="0 0 0"/>
    <parent link="body"/>
    <child link="left_upper"/>
  </joint>

  <!-- upper link -->
  <link name="left_upper">
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://human_model/meshes/stl/left_upper.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>

    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://human_model/meshes/stl/left_upper.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"/>
    </visual>

    <inertial>
      <mass value="1e-5" />
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6" />
    </inertial>
  </link>

  <!-- lower to upper -->
  <joint name="left_elbow" type="fixed">
    <origin xyz="-0.01 0.015 -0.368" rpy="0 0 0"/>
    <parent link="left_upper"/>
    <child link="left_lower"/>
  </joint>

  <!-- lower link -->
  <link name="left_lower">
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://human_model/meshes/stl/left_lower.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>

    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://human_model/meshes/stl/left_lower.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"/>
    </visual>

    <inertial>
      <mass value="1e-5" />
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6" />
    </inertial>
  </link>

  <!-- hand to lower -->
  <joint name="left_wrist" type="fixed">
    <origin xyz="0.05 0 -0.29" rpy="0 0 0"/>
    <parent link="left_lower"/>
    <child link="left_hand"/>
  </joint>

  <!-- hand link -->
  <link name="left_hand">
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://human_model/meshes/stl/left_hand.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>

    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://human_model/meshes/stl/left_hand.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"/>
    </visual>

    <inertial>
      <mass value="1e-5" />
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6" />
    </inertial>
  </link>

  <!-- =============================RIGHT ARM========================================== -->
  <!-- upper to body -->
  <joint name="right_shoulder" type="fixed">
    <origin xyz="0.015 -0.225 0.46" rpy="0 0 0"/>
    <parent link="body"/>
    <child link="right_upper"/>
  </joint>

  <!-- upper link -->
  <link name="right_upper">
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://human_model/meshes/stl/right_upper.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>

    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://human_model/meshes/stl/right_upper.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"/>
    </visual>

    <inertial>
      <mass value="1e-5" />
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6" />
    </inertial>
  </link>

  <!-- lower to upper -->
  <joint name="right_elbow" type="fixed">
    <origin xyz="-0.01 -0.015 -0.368" rpy="0 0 0"/>
    <parent link="right_upper"/>
    <child link="right_lower"/>
  </joint>

  <!-- lower link -->
  <link name="right_lower">
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://human_model/meshes/stl/right_lower.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>

    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://human_model/meshes/stl/right_lower.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"/>
    </visual>

    <inertial>
      <mass value="1e-5" />
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6" />
    </inertial>
  </link>

  <!-- hand to lower -->
  <joint name="right_wrist" type="fixed">
    <origin xyz="0.05 0 -0.29" rpy="0 0 0"/>
    <parent link="right_lower"/>
    <child link="right_hand"/>
  </joint>

  <!-- hand link -->
  <link name="right_hand">
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://human_model/meshes/stl/right_hand.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>

    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://human_model/meshes/stl/right_hand.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="grey"/>
    </visual>

    <inertial>
      <mass value="1e-5" />
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6" />
    </inertial>
  </link>

  <!-- Colors -->
  <material name="blue">
    <color rgba="0.1 0.1 0.9 1"/>
  </material>

  <material name="grey">
    <color rgba="0.8 0.8 0.8 1"/>
  </material>

  <material name="red">
    <color rgba="0.9 0.1 0.1 1"/>
  </material>

  <material name="green">
    <color rgba="0.2 0.9 0.2 1"/>
  </material>

</robot>